src/ploigos_step_runner/step_implementers/static_code_analysis/sonarqube.py

```diff
@@ -150,31 +153,30 @@ def _required_config_or_result_keys():
     def _validate_required_config_or_previous_step_result_artifact_keys(self):
         """Validates that the required configuration keys or previous step result artifacts
         are set and have valid values.
-
         Validates that:
         * required configuration is given
-        * either both username and password are set or neither.
-
+        * either both username and password are set or neither if no token.
```

Comment:
Cosmetic, but the wording here is a bit confusing. I might go with something to indicate the three cases, like "either username and password are set but not token, token is set but not username and password, or none are set" (probably a much cleaner way to do this, but hopefully this conveys the idea)